python/hsfs/core/online_store_sql_engine.py

```diff
@@ -557,10 +557,11 @@ async def _get_connection_pool(self, default_min_size: int) -> None:
 
     async def _query_async_sql(self, stmt, bind_params):
         """Query prepared statement together with bind params using aiomysql connection pool"""
-        if self._connection_pool is None:
-            await self._get_connection_pool(
-                len(self._prepared_statements[self.SINGLE_VECTOR_KEY])
-            )
+        # create connection pool
+        await self._get_connection_pool(
```

Comment:
how much will this affect the latency of get_feature_vector if we create a new engine for each request?

Reply:
trying it out using sim search test i didn't see any notable change.

I also tried to move the call `self._get_connection_pool` to here: https://github.com/logicalclocks/hopsworks-api/blob/6ef3051bfd76179f6954573d8c5e8dcf2bd6b76a/python/hsfs/core/online_store_sql_engine.py#L600 but it didn't work, if i tried to close it it failed and table remained locked (not sure why).

Reply:
ok, I will approve it for now as we need the fix asap, but I guess there should be some ways to reuse the connection pool.